File: source/ResourceLibrary/Models/Domain/FolderTypeModel.vb

Namespace Models.Domain
    '
    Public Class FolderTypeModel
        Public FolderID As Integer
        Public parentFolderID As Integer
        Public Name As String
        Public FullPath As String
        '
        Public hasViewAccess As Boolean                    ' has permission to view this folder (below topFolderPath)
        Public viewAccessIsValid As Boolean                 ' true when hasViewAccess is correct
        '
        Public hasModifyAccess As Boolean                  ' has permission to modify files and folders in this folder
        Public modifyAccessIsValid As Boolean              ' true when hasModifyAccess is correct
    End Class
End Namespace

File: source/ResourceLibrary/Models/Domain/FolderTypeModel.cs

namespace Contensive.Addons.ResourceLibrary.Models.Domain {
    public class FolderTypeModel {
        public int FolderID;
        public int parentFolderID;
        public string Name;
        public string FullPath;
        public bool hasViewAccess;
        public bool viewAccessIsValid;
        public bool hasModifyAccess;
        public bool modifyAccessIsValid;
    }
}
